src/GafferScene/ScenePlug.cpp

```diff
@@ -271,32 +245,49 @@ const PathMatcherDataPlug *ScenePlug::setPlug() const
 	return getChild<PathMatcherDataPlug>( 7 );
 }
 
+ScenePlug::PathScope::PathScope( const Gaffer::Context *context, const ScenePath &scenePath )
```

Comment:
I'm wondering if these should have convenience setters  (e.g. `PathScope::setPath` and `SetScope::setSetName`) as well as (or instead of) the constructor arg. If we've got the struct itself for convenience, might as well make it easier on the caller when they want to use it to scope a loop (as in `fullAttributes()` below).